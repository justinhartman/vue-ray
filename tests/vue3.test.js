/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */

import RayVue3Plugin from './../src/RayVue3';
import { FakeRay } from './TestClasses/FakeRay';
import { FakeVm } from './TestClasses/FakeVm';

let ray, fakeApp;

class FakeApp {
    constructor() {
        this.providedItems = [];
        this.config = {
            globalProperties: {
                $ray: null,
            },
            errorHandler: null,
        };

        this.mixins = {};
        this.mixins.methods = {
            $ray: null,
        };
    }

    provide(name, options) {
        this.providedItems.push(name);
    }

    getProvided() {
        return this.providedItems;
    }

    getLastProvided() {
        return this.getProvided().slice(0).pop();
    }

    mixin(obj) {
        this.mixins = Object.assign({}, this.mixins, obj);
    }
}

beforeEach(() => {
    ray = new FakeRay();
    fakeApp = new FakeApp();
});

describe('Vue 3 Ray Plugin:', () => {
    test('it installs successfully', () => {
        RayVue3Plugin.install(fakeApp, {});

        expect(fakeApp.getLastProvided()).toBe('ray');
        expect(fakeApp.getProvided().length).toBe(1);
        //expect(fakeApp.config.globalProperties['$ray']).not.toBe(null);
        expect(fakeApp.mixins.methods.$ray).not.toBe(null);

        if (fakeApp.mixins.methods.$ray !== null) {
            const testRay = fakeApp.mixins.methods.$ray;

            expect(typeof testRay).not.toBe('undefined');
            expect(testRay().constructor.name).toBe('VueRay');
        }
    });

    test('it installs successfully with the interceptErrors option', () => {
        RayVue3Plugin.install(fakeApp, { interceptErrors: true });

        expect(fakeApp.getLastProvided()).toBe('ray');
        expect(fakeApp.getProvided().length).toBe(1);
        expect(fakeApp.mixins.methods.$ray).not.toBe(null);
        expect(fakeApp.config.errorHandler).not.toBeNull();
        expect(typeof fakeApp.config.errorHandler).toBe('function');

        const fakeVm = new FakeVm(ray);
        fakeApp.config.errorHandler({ stack: 'test error' }, fakeVm);

        expect(ray.payloads.length).toBe(3);
        expect(ray.payloads[0].content).toContain('test error');
        expect(ray.payloads[1].content).toBe('small');
        expect(ray.payloads[2].content).toBe('red');
    });
});
