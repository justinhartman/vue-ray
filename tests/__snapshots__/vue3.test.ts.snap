// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vue 3 Mixin: it conditionally displays the beforeMount event 1`] = `
Array [
  Object {
    "content": Array [
      "component before-mount: <code>test</code>",
      "filename: <code>&lt;project root&gt;/test.js</code>",
    ],
    "label": undefined,
    "name": "table",
  },
]
`;

exports[`Vue 3 Mixin: it conditionally displays the created & mounted events 1`] = `
Array [
  Object {
    "content": Array [
      "component created: <code>test</code>",
      "filename: <code>&lt;project root&gt;/test.js</code>",
    ],
    "label": undefined,
    "name": "table",
  },
  Object {
    "content": Array [
      "component mounted: <code>test</code>",
      "filename: <code>&lt;project root&gt;/test.js</code>",
    ],
    "label": undefined,
    "name": "table",
  },
]
`;

exports[`Vue 3 Mixin: it conditionally displays the unmounted event 1`] = `
Array [
  Object {
    "content": Array [
      "component unmounted: <code>test</code>",
      "filename: <code>&lt;project root&gt;/test.js</code>",
    ],
    "label": undefined,
    "name": "table",
  },
]
`;

exports[`Vue 3 Mixin: it conditionally displays the updated event 1`] = `
Array [
  Object {
    "content": Array [
      "component updated: <code>test</code>",
      "filename: <code>&lt;project root&gt;/test.js</code>",
    ],
    "label": undefined,
    "name": "table",
  },
]
`;
